client/handle_http.go

```diff
@@ -2381,6 +2421,94 @@ func (te *TransferEngine) walkDirUpload(job *clientTransferJob, transfers []tran
 	return err
 }
 
+// This function performs the ls command by walking through the specified directory and printing the contents of the files
+func listHttp(ctx context.Context, remoteObjectUrl *url.URL, directorUrl string, namespace namespaces.Namespace, token string, flagOptions map[string]bool) (err error) {
+	// Get our directory listing host
+	dirListHost, err := getDirListHost(ctx, remoteObjectUrl, namespace, directorUrl)
+	if err != nil {
+		return
+	}
+	log.Debugln("Dir list host: ", dirListHost.String())
+
+	auth := &bearerAuth{token: token}
+	client := gowebdav.NewAuthClient(dirListHost.String(), auth)
+
+	// XRootD does not like keep alives and kills things, so turn them off.
+	transport := config.GetTransport()
+	client.SetTransport(transport)
+	remotePath := remoteObjectUrl.Path
+
+	infos, err := client.ReadDir(remotePath)
+	if err != nil {
+		// Check if we got a 404:
+		if gowebdav.IsErrNotFound(err) {
+			err = errors.Wrap(err, "file not found")
+		} else if gowebdav.IsErrCode(err, 500) {
+			// If we get an error code 500 (internal server error), we should check if the user is trying to ls on a file
+			info, err := client.Stat(remotePath)
+			if err != nil {
+				return errors.Wrap(err, "failed to stat remote path")
+			}
+			// If the path leads to a file and not a directory, just print the filename
+			if !info.IsDir() {
+				// Note: for some reason info.Name() returns nothing, so just printing base of the provided path since it should be the same
+				fmt.Println(path.Base(remotePath))
+				return nil
+			}
+		}
+		// Otherwise, a different error occurred and we should return it
+		return errors.Wrap(err, "failed to read remote directory")
+	}
+
+	// if the -L flag was set, we print more information
+	if flagOptions["long"] {
+		w := tabwriter.NewWriter(os.Stdout, 1, 2, 10, ' ', tabwriter.TabIndent|tabwriter.DiscardEmptyColumns)
```

Comment:
The output of the API function should be a data structure - or a channel of data structures - describing the listed object.

Writing to stdout should be done in the client, not the library.